ui/.gitignore

```diff
@@ -8,3 +8,4 @@ dist
 
 src/api/
 src/routeTree.gen.ts
+build/
```

Comment:
Commit message nits:
- "openApi" -> "OpenAPI"
- "when running theformatting checks" -> "when running the `format:check` script"
- Put `.gitignore` into backticks to improve formatting.

Reply:
You could just add the file to the `.prettierignore` instead to avoid the "duplication" in the `.gitignore`.

Reply:
IMO the logic is more sound if `.gitignore` includes all gitignored files and Prettier ignores files that are gitignored based on that. For files that are tracked in Git but for some reason should still not be formatted, like lockfiles, `.prettierignore` makes sense.

Reply:
Does adding it to `.gitignore` actually work? Because in 1320b42a954b59c0958559bbafe776c70745248a I have added the `--ignore-file .prettierignore` option so that the generated API code is still formatted even though it is in `.gitignore`.
And I tend to agree with the creator of https://github.com/prettier/prettier/issues/15438 that using `.gitignore` for Prettier is problematic, because it mixes the configuration of different tools for different purposes, potentially leading to conflicts.

Reply:
I believe you're right, Martin, we did discuss this earlier, and I actually would prefer your proposal over the `.gitignore` one.

Reply:
> Does adding it to `.gitignore` actually work? Because in [1320b42](https://github.com/eclipse-apoapsis/ort-server/commit/1320b42a954b59c0958559bbafe776c70745248a) I have added the `--ignore-file .prettierignore` option so that the generated API code is still formatted even though it is in `.gitignore`.

It would work - by setting the `--ignore-file` the default ignore files (`.gitignore,.prettierignore`) are not used, so `.gitignore` is not used for that command at all.

> And I tend to agree with the creator of [prettier/prettier#15438](https://github.com/prettier/prettier/issues/15438) that using `.gitignore` for Prettier is problematic, because it mixes the configuration of different tools for different purposes, potentially leading to conflicts.

Yep, looks problematic, though it works as it works. We could surely use `--ignore-file .prettierignore` with all formatting calls and if we want to save computer cycles, copy `node_modules` etc. to `.prettierignore`.

> Mikko, do you succumb?

Sure, no strong opinions, just feels more logical to me to use `.prettierignore` for non-gitignored files, but totally fine by me either way. :)

Reply:
I decided to redo this the way Martin suggested; now done.